spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/TaskSchedulerWrapper.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2012-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.commons.util;
+
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.util.Assert;
+
+/**
+ * Wrapper that downstream projects can use to keep {@link ThreadPoolTaskScheduler} local.
+ *
+ * Implementation taken from Spring Vault.
+ *
+ */
+public class TaskSchedulerWrapper implements InitializingBean, DisposableBean {
+
+	private final ThreadPoolTaskScheduler taskScheduler;
```

Comment:
Maybe make it either generic <? extends TaskScheduler> or use TaskScheduler interface